### PR TITLE
Reworks blood regeneration into something a bit more tame

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
Every mob now has two blood pools, one that is your normal blood pool, works like normal, but you normally get blood back into it directly, instead it's put int your other blood pool, which slowly becomes your normal blood, this is intended to deal with stuff like being able to regenerate blood stupidly quickly and being able to use it for stuff, also this will fix the saline glucose potassium water exploit for getting blood

## Why It's Good For The Game
Fixes a lot of cheesy issues with blood

## Changelog
:cl:
tweak: Attempts to refill a mobs blood are now gradual, but still protect against low blood effects, but don't allow it being used for things that use blood
del: Nonslimes and nonvampires will no longer be able to increase their blood to stupid heights
/:cl:

### CODE REVIEW

**Comment by HeroWithYay** [comment]
You gonna put this on a wiki somewhere? Because this seems unnecessarily complex.

**Comment by Arturlang** [comment]
@HeroWithYay All this means practically is you're going to have to wait for a bit before being able to donate blood or be able to give practical amounts of blood to bloodsuckers if you use ways of regenerating blood, the blood buffer thing otherwise will stop you from suffering as if you were low on blood

**Comment by Poojawa** [comment]
This + Armor thresholds for actual blood loss (instead of it having just been a shitty hard delete), was the ultimate goal of making synthetic blood replacement actually worth a godamn. 

Good work.

**Comment by The-Sun-In-Splendour** [comment]
I don't remember seeing anyone asking tho?

**Comment by The-Sun-In-Splendour** [comment]
> 
> 
> This + Armor thresholds for actual blood loss (instead of it having just been a shitty hard delete), was the ultimate goal of making synthetic blood replacement actually worth a godamn.
> 
> Good work.

Synth blood is already worth a goddamn. I can literally just replace every blood bags in medbay with synth blood because they spawn roundstart for some god awful reason, while they also generate MORE synthblood than blood consumed for some awful reason. You can literally transform every blood bag into essentially omni-specie O- roundstart.

**Comment by Poojawa** [comment]
Good lord, someone's grumpy.

Before synthetics, there was no medbay specific,  useful means of blood transfusions. You either ate entire ingots of iron (without dying of OD, because there wasn't any other way to replace blood since sal/glu wasn't reliable) or hoped the bartender wasn't balls deep in someone's ass in dorms to mix you a drink.

File: code/modules/mob/living/blood.dm [hunk 1]
```diff
@@ -39,29 +39,34 @@
...
-					nutrition_ratio = 0.6
-				if(NUTRITION_LEVEL_FED to NUTRITION_LEVEL_WELL_FED)
-					nutrition_ratio = 0.8
-				else
-					nutrition_ratio = 1
-			if(satiety > 80)
-				nutrition_ratio *= 1.25
-			adjust_nutrition(-nutrition_ratio * HUNGER_FACTOR)
-			blood_volume = min(BLOOD_VOLUME_NORMAL, blood_volume + 0.5 * nutrition_ratio)
+			if(integrating_blood) //Do we have blood to normalize in our system?
```
Comment (Putnam3145) [inline code/modules/mob/living/blood.dm]: the fact that this is after the integrating_blood-- line means that, if someone has 1 integrating_blood, they won't actually integrate any blood. Is this intentional?

Comment (Arturlang) [reply code/modules/mob/living/blood.dm]: noted and fixed, hopefully

**Comment by The-Sun-In-Splendour** [comment]
> 
> 
> Good lord, someone's grumpy.
> 
> Before synthetics, there was no medbay specific, useful means of blood transfusions. You either ate entire ingots of iron (without dying of OD, because there wasn't any other way to replace blood since sal/glu wasn't reliable) or hoped the bartender wasn't balls deep in someone's ass in dorms to mix you a drink.

"Someone is grumpy" I have never said synth blood needed to be removed.
I'm just saying that synth blood "not being worth a damn" is ignorant.

**Comment by Arturlang** [comment]
TODO, make salglu remove the blood it's given, properly

**Comment by Gboster-0** [comment]
kinda a pain that plasma in the air is preety much useless for getting enough blood for clones as a slimeperson.

**Comment by Arturlang** [comment]
Pretty much ready now

**Comment by Arturlang** [comment]
This does now somewhat buff salgul's blood regenration

**Comment by Arturlang** [comment]
Salglu broke even though i througoughly tested it

**Pull request merged**